test/tools/ossfuzz/abiV2ProtoFuzzer.cpp

```diff
@@ -31,20 +31,77 @@ using namespace std;
 
 namespace
 {
-	static size_t const expectedOutputLength = 32;
-	// Expected output is a uint256 whose value is decimal 1000 or hex 01E8.
-	static uint8_t const expectedOutput[expectedOutputLength] = {
-		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, u'\x01', u'\xe8'
+/// Test function returns a uint256 value
+static size_t const expectedOutputLength = 32;
+/// Expected output value is decimal 1000 or hex 03E8
+static uint8_t const expectedOutput[expectedOutputLength] = {
+	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, u'\x03', u'\xe8'
+};
+
+/// Compares the contents of the memory address pointed to
+/// by `_result` of `_length` bytes to the expected output.
+/// Returns true if `_result` matches expected output, false
+/// otherwise.
+bool isOutputExpected(uint8_t const* _result, size_t _length)
+{
+	if (_length != expectedOutputLength)
+		return false;
+
+	return (memcmp(_result, expectedOutput, expectedOutputLength) == 0);
+}
+
+/// Accepts a reference to an evmc_message and a user-specified input
+/// and zero initializes all fields except gas and input fields.
+/// The gas field is set to the maximum permissible value so that we
+/// don't run into out of gas errors. The input field is copied from
+/// user input.
+void initializeMessage(evmc_message& _msg, dev::bytes const& _input)
```

Comment:
I think it would be better to make this function return an `evmc_message`

Reply:
Okay, I guess my assumption here was that return by value would result in a deep-copy of all `evmc_message` fields and we can save cycles by passing a reference and populating it inside.